### ISSUE TITLE
Building a sparse neural network

### ISSUE THREAD

**Issue by qxzhong** [rstudio/keras#1118]
Hi,

I am building a sparse neural network (NN) with Keras, but I didn't know the relevant code. Would you please give me any ideas about it? The sparse NN means that a large number of weights (e.g. 50 percent of weights) are set to be zero. Those weights with (initial) zeros do not get learnt during learning.  It is not the dropout, where the neurons are removed during the training, but the final weights are not equal to zero. It is still a fully connected network. 

Many thanks,
QZ
